pkg/kubectl/resource/builder.go

```diff
@@ -806,12 +806,13 @@ func HasNames(args []string) (bool, error) {
 
 // MultipleTypesRequested returns true if the provided args contain multiple resource kinds
 func MultipleTypesRequested(args []string) bool {
+	if len(args) == 1 && args[0] == "all" {
+		return true
```

Comment:
This check is somewhat misleading. In resource/builder.go, the call to `ReplaceAliases` replaces `all` with all supported resources [1], and then `ResourceTypeOrNameArgs` will split it into a slice of string [2]

1.`pods,replicationcontrollers,services,statefulsets,horizontalpodautoscalers,jobs,deployments,replicasets`
2.`[]string{"pods,replicationcontrollers,services,statefulsets,horizontalpodautoscalers,jobs,deployments,replicasets"}`

This means that `all` won't be passed in as `args[0]`.

The unit test needs to be updated as well.